### PR TITLE
BO flags cleanup

### PR DESCRIPTION
**Pull request by digetx**
Let's fix a possibility of getting a BO with a mismatched flags from cache and maintain flags state after updating them in set_flags().

### CODE REVIEW

**Comment by digetx** [comment]
Oh wait, we have special IOCTL's for tiling.

**Comment by digetx** [comment]
However, we should ignore the tiling flag.

**Comment by digetx** [comment]
Hmm, I just realized that seems VG_BO_RELEASE / VG_BO_OBTAIN doesn't really work. In commit https://github.com/grate-driver/libdrm/pull/17/commits/7e895d685a4afc3bc3d6a24d870e353833859ab3 BO is adjusted before VG_BO_OBTAIN() invocation and I think valgrind should complain about it, it's not.

**Comment by digetx** [comment]
Ah, so VG_BO_*() actually disable/enable accesses to BO, but in the same time disable/enable error reporting of the wrong accesses, which is kinda strange.

**Comment by digetx** [comment]
Yeah, enabling valgrind reporting would be a bit messy due to adjusting of neighbor BO's during the removal from the cache 'bucket' linked list.

**Comment by digetx** [comment]
However, a quick solution is to *not* protect the cache related BO fields, I'll implement it after this PR will go in.

File: tegra/tegra_bo_cache.c [hunk 1]
```diff
@@ -148,15 +152,29 @@ static struct drm_tegra_bo *find_in_bucket(struct drm_tegra_bo_bucket *bucket,
 	 */
 	pthread_mutex_lock(&table_lock);
 	if (!DRMLISTEMPTY(&bucket->list)) {
-		bo = DRMLISTENTRY(struct drm_tegra_bo, bucket->list.next,
+		tmp = DRMLISTENTRY(struct drm_tegra_bo, bucket->list.next,
```
Comment (kusma) [inline tegra/tegra_bo_cache.c]: Maybe move the definition of `tmp`, `tiling` and `err` here also?

**Comment by digetx** [comment]
I'm going to factor out the BO flags resetting in the followup patch (BO valgrind-protection enabling), so literally these find_in_bucket() changes will be reverted.

File: tegra/tegra.c [hunk 2]
```diff
@@ -156,6 +156,9 @@ int drm_tegra_bo_new(struct drm_tegra_bo **bop, struct drm_tegra *drm,
 	if (!drm || size == 0 || !bop)
 		return -EINVAL;
 
+	if (flags & ~DRM_TEGRA_GEM_FLAGS)
```
Comment (digetx) [inline tegra/tegra.c]: I just noticed that this is incorrect, DRM_TEGRA_GEM_FLAGS have nothing to do with the BO creation flags.

**Comment by kusma** [comment]
There's still some work to be done on this one, right?

**Comment by digetx** [comment]
This PR isn't actual anymore, I have move BO flags resetting to https://github.com/grate-driver/libdrm/pull/20.

**Pull request closed without merging**